usvm-ts/src/main/kotlin/org/usvm/machine/expr/TSExprResolver.kt

```diff
@@ -0,0 +1,555 @@
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsCommaExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsDeleteExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsDivExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsExpExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsGtEqExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsGtExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsInExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsInstanceCallExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsInstanceOfExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsLeftShiftExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsLengthExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsLtEqExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsLtExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsMulExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsNegExpr): UExpr<out USort>? {
+        if (expr.type is EtsNumberType) {
```

Comment:
Maybe move to unary operator?

Reply:
Added todo, will move when we support all unary operators